components/kyma-environment-broker/cmd/broker/main.go

```diff
@@ -922,6 +917,11 @@ func NewKymaOrchestrationProcessingQueue(ctx context.Context, db storage.BrokerS
 			step:   upgrade_kyma.NewApplyClusterConfigurationStep(db.Operations(), db.RuntimeStates(), reconcilerClient),
 			cnd:    upgrade_kyma.ForKyma2,
 		},
+		{
+			weight: 11,
+			step:   upgrade_kyma.NewCheckClusterConfigurationStep(db.Operations(), reconcilerClient, 15*time.Minute),
+			cnd:    upgrade_kyma.ForKyma2,
```

Comment:
for some reason, this step was before calling reconciler API. I think the check should be right after the reconciler call.

Reply:
It sounds correct what you are saying.

However, I remember looking over this part together with @ksputo - wasn't there a specific reason for checking the cluster config earlier?

Reply:
I guess we will find out soon, however, this was preventing running upgrade on kyma2 whether migrated or not. So we wouldn't be able to upgrade even with service catalog from 2.0.0 to 2.0.1.

Reply:
even after followup inspection, I couldn't figure out a reason why check should be the first step in the queue, especially if it is the only step that marks the operation as "finished".

Thank you for the heads up @voigt, I will make sure to discuss with @ksputo and also keep an eye for potential bug reports in this area.